Sources/Moya/TargetType.swift

```diff
@@ -38,32 +32,36 @@ public extension TargetType {
     }
 }
 
-/// Represents a type of upload task.
-public enum UploadType {
+/// Represents an HTTP task.
+public enum Task {
 
-    /// Upload a file.
-    case file(URL)
+    /// A request with no additional data.
+    case requestPlain
 
-    /// Upload "multipart/form-data"
-    case multipart([MultipartFormData])
-}
+    /// A requests body set with data.
+    case requestData(Data)
 
-/// Represents a type of download task.
-public enum DownloadType {
+    /// A requests body set with encoded parameters.
+    case requestParameters(parameters: [String: Any], encoding: ParameterEncoding)
 
-    /// Download a file to a destination.
-    case request(DownloadDestination)
-}
+    /// A requests body set with data, combined with url parameters.
+    case requestCompositeData(bodyData: Data, urlParameters: [String: Any])
 
-/// Represents an HTTP task.
-public enum Task {
+    /// A requests body set with encoded parameters combined with url parameters.
+    case requestCompositeParameters(bodyParameters: [String: Any], bodyEncoding: ParameterEncoding, urlParameters: [String: Any])
```

Comment:
Yeah, we can achieve this with `requestCompositeData`. It may not be as clean, but I think it's better than undefined behavior. In my experience, if it's possible for someone to make a mistake -- someone will. It could be as simple as using the wrong encoding w/ auto-complete.

I'm fine with removing it for now. We can start simple and then expand the API from there based on user feedback. Another suggestion would be to throw an error in this situation. The previous API returned nil and ignored encoding errors, but now we have a lot going on here.